plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java

```diff
@@ -624,22 +636,51 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
         }
 
         String typeName = typeHandle.getJdbcTypeName().get();
-        if (typeName.equalsIgnoreCase("tinyint unsigned")) {
-            return Optional.of(smallintColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("smallint unsigned")) {
-            return Optional.of(integerColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("int unsigned")) {
-            return Optional.of(bigintColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("bigint unsigned")) {
-            return Optional.of(decimalColumnMapping(createDecimalType(20)));
+        if (UNSIGNED_TYPE_REGEX.matcher(typeName).matches()) {
+            switch (typeHandle.getJdbcType()) {
+                case Types.BIT:
+                    return Optional.of(booleanColumnMapping());
+
+                case Types.TINYINT:
+                    return Optional.of(smallintColumnMapping());
+
+                case Types.SMALLINT:
+                    return Optional.of(integerColumnMapping());
+
+                case Types.INTEGER:
+                    return Optional.of(bigintColumnMapping());
+
+                case Types.BIGINT:
+                    return Optional.of(decimalColumnMapping(createDecimalType(20)));
+            }
         }
 
         return Optional.empty();
     }
 
+    private static LongReadFunction memsqlTimeReadFunction(TimeType timeType)
+    {
+        requireNonNull(timeType, "timeType is null");
+        checkArgument(timeType.getPrecision() <= 9, "Unsupported type precision: %s", timeType);
+        return (resultSet, columnIndex) -> {
+            // SingleStore stores incorrect results when values are outside supported range, so we must validate them
```

Comment:
comment in not entirely true. It stores "correct" values, the driver returns incorrect values when using LocalTime.

```suggestion
            // SingleStore JDBC driver wraps time to be within LocalTime range via getObject(i, LocalTime.class) which returns values different from what was actually stored - so we verify this
```

Reply:
👍 
```
// SingleStore JDBC driver wraps time to be within LocalTime range, which results in values which differ from what is stored
```
(just to shorten it / "values which differ from [storage]" is the reason).